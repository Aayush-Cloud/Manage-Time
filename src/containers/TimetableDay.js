// @flow
import React from 'react';
import { connect } from 'react-redux';

import CourseAppointment from '../components/CourseAppointment';
import DayBorder from '../components/DayBorder';

/* eslint-disable react/no-unused-prop-types, because of false positive */
type Props = {
  courses: CoursesState,
  day: string,
  appointmentsOfDay: Array<Appointment>,
};
/* eslint-enable react/no-unused-prop-types */

function renderCoursesOfDay(props: Props) {
  const { courses, appointmentsOfDay } = props;
  return appointmentsOfDay.map(appointment => {
    const course = courses[appointment.course];
    if (!course) {
      throw new Error(
        `The course of appointment ${appointment.id} does not exist`,
      );
    }
    return (
      <CourseAppointment
        name={course.name}
        starttime={appointment.starttime}
        endtime={appointment.endtime}
        location={appointment.location}
        type={appointment.type}
        key={appointment.id}
      />
    );
  });
}

// exported for tests
export function TimetableDay(props: Props) {
  const { appointmentsOfDay, day } = props;
  if (appointmentsOfDay.length !== 0) {
    return <DayBorder date={day}>{renderCoursesOfDay(props)}</DayBorder>;
  }
  return null;
}

function mapStateToProps(state) {
  return {
    courses: state.courses,
  };
}

export default connect(mapStateToProps)(TimetableDay);
