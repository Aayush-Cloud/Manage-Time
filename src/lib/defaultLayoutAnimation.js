// @flow
import { LayoutAnimation } from 'react-native';

const animation = LayoutAnimation.create(80, 'linear', 'opacity');

export default animation;
