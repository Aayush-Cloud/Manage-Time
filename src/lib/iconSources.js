// @flow
import Icon from 'react-native-vector-icons/MaterialIcons';
import { toPairs } from 'ramda';
import { metrics } from '../themes';

const icons = {
  close: {
    size: metrics.icons.small,
  },
};

const iconSources = {
  close: undefined,
};

// $FlowFixMe
const iconsLoaded = new Promise(resolve => {
  const iconNamePropsPairs = toPairs(icons);
  Promise.all(
    iconNamePropsPairs.map(([iconName, iconProps]) =>
      Icon.getImageSource(iconName, iconProps.size),
    ),
  ).then(sources => {
    iconNamePropsPairs.forEach(([iconName], index) => {
      iconSources[iconName] = sources[index];
    });
    resolve(true);
  });
});

export { iconSources, iconsLoaded };
