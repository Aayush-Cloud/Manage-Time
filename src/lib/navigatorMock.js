// @flow
export default {
  push: jest.fn(),
  pop: jest.fn(),
  popToRoot: jest.fn(),
  resetTo: jest.fn(),
  showModal: jest.fn(),
  dismissModal: jest.fn(),
  dismissAllModals: jest.fn(),
  showLightBox: jest.fn(),
  dismissLightBox: jest.fn(),
  handleDeepLink: jest.fn(),
  setOnNavigatorEvent: jest.fn(),
  setButtons: jest.fn(),
  setTitle: jest.fn(),
  toggleNavBar: jest.fn(),
};
