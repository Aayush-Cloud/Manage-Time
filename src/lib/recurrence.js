// @flow
export const WEEKLY = 'WEEKLY';
export const BIWEEKLY = 'BIWEEKLY';
