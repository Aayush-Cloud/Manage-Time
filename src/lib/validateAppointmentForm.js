// @flow
import i18n from 'react-native-i18n';

export default function validate(values: any) {
  const errors = {};
  // starttime and endtime errors are displayed in same <Text />
  // so both errors are just put into errors.endtime
  if (values.starttime.isAfter(values.endtime, 'minutes')) {
    errors.endtime = i18n.t('endtime-greater-than-starttime');
  }
  if (values.startdate.isAfter(values.enddate, 'minutes')) {
    errors.enddate = i18n.t('enddate-greater-than-startdate');
  }
  return errors;
}
