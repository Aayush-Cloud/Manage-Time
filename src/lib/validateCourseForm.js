// @flow
import i18n from 'react-native-i18n';

export default function validate(values: any) {
  const errors = {};
  if (!values.name) {
    errors.name = i18n.t('title-is-required');
  }
  if (!values.appointments || !values.appointments.length) {
    errors.appointments = { _error: i18n.t('one-appointment-required') };
  }
  return errors;
}
