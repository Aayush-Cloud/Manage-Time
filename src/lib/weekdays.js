// @flow

export const MON = 'mon';
export const TUE = 'tue';
export const WED = 'wed';
export const THU = 'thu';
export const FRI = 'fri';
export const SAT = 'sat';
export const SUN = 'sun';

const weekdays = [SUN, MON, TUE, WED, THU, FRI, SAT];
export default weekdays;
