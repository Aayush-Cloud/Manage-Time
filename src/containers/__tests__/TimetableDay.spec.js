// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import mockdate from 'mockdate';
import { WEEKLY } from '../../lib/recurrence';
import { TimetableDay } from '../TimetableDay';

mockdate.set(moment('01-01-2017', 'MM-DD-YYYY').startOf('day'));

const courses: CoursesState = {
  course1: {
    id: 'course1',
    name: 'A course',
    appointments: ['app1'],
  },
  course2: {
    id: 'course2',
    name: 'Another course',
    appointments: ['app2'],
  },
};

const appointmentsOfDay: Array<Appointment> = [
  {
    id: 'app1',
    course: 'course1',
    startdate: moment(),
    enddate: moment().add(3, 'weeks'),
    starttime: moment(),
    endtime: moment().add(1, 'hour'),
    recurrence: WEEKLY,
  },
  {
    id: 'app2',
    course: 'course2',
    startdate: moment().add(1, 'day'),
    enddate: moment().add(3, 'weeks'),
    starttime: moment().add(1, 'hour'),
    endtime: moment().add(2, 'hour'),
    recurrence: WEEKLY,
  },
];

const props = {
  courses,
  day: 'mon',
  appointmentsOfDay,
};

describe('TimetableDay', () => {
  it('renders a CourseAppointment for each element of props.appointmentsOfDay', () => {
    const tree = renderer.create(<TimetableDay {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
