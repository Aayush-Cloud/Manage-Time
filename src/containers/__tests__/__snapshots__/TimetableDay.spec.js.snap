// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimetableDay renders a CourseAppointment for each element of props.appointmentsOfDay 1`] = `
<View>
  <View
    style={
      Object {
        "height": 48,
        "justifyContent": "center",
        "marginLeft": 16,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "Helvetica Neue",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      mon
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "shadowOffset": Object {
            "height": 1.2,
          },
          "shadowOpacity": 0.183,
          "shadowRadius": 1.08,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 72,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "justifyContent": "center",
              "width": 72,
            }
          }
        >
          <View
            style={
              Object {
                "justifyContent": "center",
                "left": 16,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                ]
              }
            >
              0:00
            </Text>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgba(0, 0, 0, 0.54)",
                  "height": 1.5,
                  "marginVertical": 4,
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                ]
              }
            >
              1:00
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#00796B",
                "fontFamily": "Helvetica Neue",
                "fontSize": 20,
                "fontWeight": "400",
              }
            }
          >
            A course
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 1,
              "marginHorizontal": 0,
            },
            Object {
              "height": 0.5,
              "marginHorizontal": 8,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 72,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "justifyContent": "center",
              "width": 72,
            }
          }
        >
          <View
            style={
              Object {
                "justifyContent": "center",
                "left": 16,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                ]
              }
            >
              1:00
            </Text>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgba(0, 0, 0, 0.54)",
                  "height": 1.5,
                  "marginVertical": 4,
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                ]
              }
            >
              2:00
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#00796B",
                "fontFamily": "Helvetica Neue",
                "fontSize": 20,
                "fontWeight": "400",
              }
            }
          >
            Another course
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 1,
              "marginHorizontal": 0,
            },
            Object {
              "height": 0.5,
              "marginHorizontal": 8,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
